x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+      buttonContent={
+        <FormattedMessage
+          id="xpack.fleet.settings.editOutputFlyout.advancedOptionsToggleLabel"
+          defaultMessage="Advanced options"
+        />
+      }
+    >
+      <>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...maxBatchBytes.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.maxBatchSizeDescriptionLabel"
+              defaultMessage="Maximum Batch Size"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...maxBatchBytes.props} placeholder="Batching Bytes" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.maxBatchSizeDescription"
+                  defaultMessage="Data will be sent to the output when the agent has events which total larger then this configured maximum."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...queueFlushTimeout.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutLabel"
+              defaultMessage="Flush Timeout"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...queueFlushTimeout.props} placeholder="Seconds" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutDescription"
+                  defaultMessage="Upon expiry the output queue is flushed and data is written to the output."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...memQueueEvents.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.memQueueEventsLabel"
+              defaultMessage="Memory Queue Events"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...memQueueEvents.props} placeholder="Events" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.memQueueEventsSizeDescription"
+                  defaultMessage="Maximum number of events that can be stored in the queue. Default is set to 4096."
```

Comment:
"Memory Queue Events" is not very specific, and I think "size" makes more sense than "events". I also think we should explain what happens when this cap is hit.

```suggestion
              defaultMessage="Memory Queue Size"
            />
          }
        >
          <EuiFlexGroup alignItems="flexStart">
            <EuiFlexItem>
              <EuiFieldNumber {...memQueueEvents.props} placeholder="Events" min={0} />
            </EuiFlexItem>
            <EuiFlexItem>
              <EuiText>
                <FormattedMessage
                  id="xpack.fleet.settings.editOutputFlyout.memQueueEventsSizeDescription"
                  defaultMessage="Maximum number of events that can be stored in the queue. Default is set to 4096. When this queue is full, new events are dropped."
```

Reply:
yes Agreed.